### PR TITLE
[ulricolo7] iP

### PR DESCRIPTION
**Pull request by ulricolo7**
# Neonbot
> “Burgers.” – What you want when you are hungry.

Neonbot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

### CODE REVIEW

File: src/main/java/Neon.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+import java.util.Objects;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.time.LocalDate;
+
+
+public class Neon {
+    private static final String NAME = "Neon";
+    private static final String DASH_BREAK = "-----------------------------------";
+    private static ArrayList<Task> list = new ArrayList<>();
```
Comment (jktang14) [inline src/main/java/Neon.java]: Perhaps could change the naming to specify what type of list this is? (Ie. taskList)

File: src/main/java/Neon.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    private static int lastListIndex = 0;
+
+    private static void greetingLine() {
```
Comment (jktang14) [inline src/main/java/Neon.java]: Perhaps you could rephrase this to sound more like a verb, for example `printGreetingLine()`

File: src/main/java/Neon.java [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+            directory.mkdir();
+            file.createNewFile();
+        }
+
+        Scanner scanner = new Scanner(new File("./data/data.txt"));
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] content = line.split("/");
+
+            boolean completed;
```
Comment (jktang14) [inline src/main/java/Neon.java]: Perhaps you could change this naming to sound more like a boolean, Eg. isCompleted

File: src/main/java/Event.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+public class Event extends Task {
+    private String startDT;
```
Comment (jktang14) [inline src/main/java/Event.java]: Perhaps you could change this naming to make it clearer what DT means

**Review by jktang14 — COMMENTED** [review]
LGTM. Most methods, variable names follow the code naming guidelines, however there are small instances of violations, correcting these should improve your code quality as a whole 👍

File: src/main/java/Neon.java [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+    }
+
+    private static void mainChat(String answer) {
+        System.out.println(DASH_BREAK);
+
+        Scanner ansObj = new Scanner(answer);
+        String first = ansObj.next();
+
+        switch (first) {
+            case "todo":
```
Comment (huangjiaxi1111) [inline src/main/java/Neon.java]: Should case be indented?

File: src/main/java/Neon.java [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+    private static void greetingLine() {
+        System.out.println(DASH_BREAK);
+        String greeting = "hello i'm " + NAME + "!\n"
+                + "what can i help you with?";
+        System.out.println(greeting);
+        System.out.println(DASH_BREAK);
+    }
+
+    private static void closingLine() {
```
Comment (huangjiaxi1111) [inline src/main/java/Neon.java:26]: Just to add on, here can change to `printClosingLine()`

File: src/main/java/Neon.java [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+
+public class Neon {
+    private static final String NAME = "Neon";
+    private static final String DASH_BREAK = "-----------------------------------";
+    private static ArrayList<Task> list = new ArrayList<>();
+    private static int lastListIndex = 0;
+
+    private static void greetingLine() {
+        System.out.println(DASH_BREAK);
```
Comment (huangjiaxi1111) [inline src/main/java/Neon.java:19]: Maybe you can write a `printDashBreak()` to abstract this out. This can make your code cleaner!

**Review by huangjiaxi1111 — COMMENTED** [review]
LGTM! Your code managed to follow most part of the coding standard.
I like how you wrapped your code to make sure that each line is not too long and used correct formats for variable, method, and class naming.
To improve, maybe you can include header comments and put classes into packages.
